app/src/main/java/com/example/esha/movieapplication/Activities/FullMovieActivity.java

```diff
@@ -0,0 +1,137 @@
+import com.example.esha.movieapplication.rest.MovieReviewList;
+import com.example.esha.movieapplication.rest.MovieVideoList;
+import com.example.esha.movieapplication.rest.Result;
+import com.example.esha.movieapplication.rest.ReviewResult;
+import com.example.esha.movieapplication.rest.VideoResults;
+import com.example.esha.movieapplication.BuildConfig;
+import com.example.esha.movieapplication.R;
+import com.example.esha.movieapplication.RetrofitManager;
+import com.example.esha.movieapplication.ReviewListAdapter;
+import java.util.ArrayList;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+
+public class FullMovieActivity extends AppCompatActivity {
+    private final String TAG = getClass().getSimpleName();
+    private RecyclerView rvReview;
+    private ReviewListAdapter reviewAdapter;
+    private ImageView imagePoster;
+    private TextView tvTitle;
+    private RatingBar ratingBar;
+    private TextView tvOverview;
+    private ImageView imgPlay;
+    private TextView displayVoteAverage;
+    private ArrayList<VideoResults> trailerList = new ArrayList<>();
+    private ArrayList<ReviewResult> reviewList = new ArrayList<>();
+    int movieId;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.full_movie);
+
+        tvTitle = findViewById(R.id.tv_title);
+        tvOverview = findViewById(R.id.tv_overview);
+        ratingBar = findViewById(R.id.rating_bar);
+        displayVoteAverage = findViewById(R.id.display_vote_average);
+        imgPlay = findViewById(R.id.img_play);
+        imagePoster = findViewById(R.id.image_poster);
+        rvReview = findViewById(R.id.rv_review);
+        LinearLayoutManager linearLayoutManager1 = new LinearLayoutManager(FullMovieActivity.this);
+        rvReview.setLayoutManager(linearLayoutManager1);
+
+        getData();
+
+        reviewAdapter = new ReviewListAdapter(FullMovieActivity.this, reviewList);
+        rvReview.setAdapter(reviewAdapter);
+    }
+
+    /**
+     * obtains trailer information of a particular movie and its reviews.
+     * displays the information in the users UI.
+     */
+
+    public void getData() {
+        Intent movieIntent = getIntent();
+        if (movieIntent == null) {
+            return;
+        }
+        final Result result = movieIntent.getParcelableExtra("result");
+        movieId = result.getId();
+
+        RetrofitManager.getInstance().getMovieTrailer(result.getId(), BuildConfig.TMDBMOVIEAPIKEY, new Callback<MovieVideoList>() {
+
+            @Override
+            public void onResponse(Call<MovieVideoList> call, Response<MovieVideoList> response) {
+
+                if (response.code() == 200) {
+                    trailerList.addAll(response.body().getResults());
+
+                } else {
+                    Toast.makeText(FullMovieActivity.this, " " + response.code(), Toast.LENGTH_SHORT).show();
```

Comment:
@Esha-Rajbhandari you are using toast multiple time so make an Util class or make an private method in this class